dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt

```diff
@@ -77,13 +78,18 @@ class DataManager(
      * Method to make the data manager add data to a data store, store metadata in Dataland and sending messages to the
      * relevant message queues
      * @param storableDataSet contains all the inputs needed by Dataland
+     * @param bypassQa whether the data should be sent to QA or not
      * @return ID of the newly stored data in the data store
      */
-    fun addDataSetToTemporaryStorageAndSendMessage(storableDataSet: StorableDataSet, correlationId: String):
+    fun addDataSetToTemporaryStorageAndSendMessage(
+        storableDataSet: StorableDataSet,
+        bypassQa: Boolean,
+        correlationId: String,
+    ):
         String {
         val dataId = generateRandomDataId()
         addDatasetToDatabase(dataId, storableDataSet, correlationId)
```

Comment:
Die Methode verwirrt mich ein bisschen.
1.) `addDatasetToDatabase()` storet doch im Grunde nur die meta info des datasets im meta storage vom backend? In dem Fall wäre der Name der Methode ziemlich untreffend, und die Methode sollte dann auch nur ein `DataMetaInfo` Objekt bekommen, statt ein gesamtes `StorableDataSet`.
2.) Falls meine Aussage zu 1.) stimmt, dann sollte die Methode auch nicht `addDataSetToTemporaryStorageAndSendMessage()` heißen, sondern eher sowas wie `storeDataSetInMemoryAndSendMessageAndPersistMetaInfo` oder so ähnlich

Außerdem aus Neugier, und um zu wissen, ob ich es richtig verstanden habe:
- Die Meta Info wird direkt in der Backend DB gespeichert (die ja im Grunde ein Meta-Speicher ist)
- Das Dataset selbst wird zur "Abholung" in memory gespeichert, und der internal storage holt es sich ab und speichert es in seiner eigenen internal_DB.  Korrekt?

Reply:
1.) Ja, haabe es in `storeMetaDataFrom` umbenannt und an der Funktionalität nichts geändert.
2.) Hab ich auch angepasst.

Die anderen beiden Punkte hast du richtig verstanden.

Reply:
Passt